EngineAPI/src/main/java/nullengine/event/world/block/cause/BlockInteractCause.java

```diff
@@ -0,0 +1,47 @@
+package nullengine.event.world.block.cause;
+
+import java.util.Optional;
+
+import nullengine.entity.Entity;
+import nullengine.event.cause.Cause;
+import nullengine.item.ItemStack;
+
+public interface BlockInteractCause extends Cause {
+    class EntityTrigger implements BlockInteractCause {
+        public static final String HIT = "entity.hit", USE = "entity.use";

```

Comment:
Again, another way to do the cause.

Some suggestions please. @Mouse0w0  @cvrunmin

Reply:
`HIT` and `USE` should move into the enum named `Action` of  `BlockInteractEvent`.

Reply:
I do prefer using class than using interface. Using interface may require creating classes implementing combinations of interfaces.

Reply:
> `HIT` and `USE` should move into the enum named `Action` of `BlockInteractEvent`. And add a field named action to `BlockInteractEvent`.

Just for extensibility requirement, if there are more than hit & use action, the enum is not a good way to do this. In this case, we just need a pivot to mark a semantic meaning of "Hit" & "Use". 

Either specific String, or immutable object (enum is a kind of immutable object without third-party extensibility) can play this role. 

I'm totally open on this.

If both of you think enum is simple and better. We just do the enum way.

Reply:
> I do prefer using class than using interface. Using interface may require creating classes implementing combinations of interfaces.

Yes, BlockInteractCause.EntityTrigger is a class & you can create instance by calling function.

All other later work, like command triggered, will be wrapped in class. Interface is just the parent.

Refer [this](https://github.com/UnknownDomainGame/NullEngine/pull/44/files#diff-d7f35bed7bf563ff0f74311b10ec7610R151)